src/main/java/dinkplugin/VersionManager.java

```diff
@@ -0,0 +1,127 @@
+package dinkplugin;
+
+import dinkplugin.util.Utils;
+import lombok.Value;
+import net.runelite.api.GameState;
+import net.runelite.client.config.ConfigManager;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.Comparator;
+import java.util.NavigableMap;
+import java.util.TreeMap;
+
+@Singleton
+public class VersionManager {
+    public static final String VERSION_CONFIG_KEY = "pluginVersion";
+
+    private static final NavigableMap<Version, String> VERSIONS = new TreeMap<>(
+        Comparator.comparingInt(Version::getMajor)
+            .thenComparingInt(Version::getMinor)
+            .thenComparingInt(Version::getPatch)
+    );
+
+    @Inject
+    private DinkPlugin plugin;
+
+    @Inject
+    private DinkPluginConfig config;
+
+    @Inject
+    private ConfigManager configManager;
+
+    @Inject
+    private SettingsManager settingsManager;
+
+    void onStart() {
+        if (config.pluginVersion().isEmpty()) {
+            if (settingsManager.hasModifiedConfig()) {
+                // Dink was already installed/used & launched for the first time since VersionManager was created
+                setStoredVersion("1.8.4");
+            } else {
+                // first time launching Dink; no chat message necessary
+                setStoredVersion(getLatestVersion());
+            }
+        }
+    }
+
+    void onGameState(GameState oldState, GameState newState) {
+        if (newState != GameState.LOGGED_IN || oldState != GameState.LOGGING_IN) {
+            // only check version when LOGGING_IN => LOGGED_IN
+            return;
+        }
+
+        Version storedVersion = Version.of(config.pluginVersion());
+        if (storedVersion == null) {
+            // this branch shouldn't be hit due to onStart()
+            return;
+        }
+
+        NavigableMap<Version, String> latestUpdates = VERSIONS.tailMap(storedVersion, false);
+        if (latestUpdates.isEmpty()) {
+            return;
+        }
+
+        Version latest = latestUpdates.lastKey();
+        setStoredVersion(latest.toString());
+        String displayVersion = latest.getPatch() == 0
+            ? String.format("%d.%d.X", latest.getMajor(), latest.getMinor()) // avoids needing changelog for each patch
+            : latest.toString();
+        plugin.addChatMessage(
+            "Updated to v" + displayVersion,
+            Utils.GREEN,
+            String.join("; ", latestUpdates.values())
+        );
```

Comment:
Might be worth delaying this message a second or so if we want people to read it, otherwise there's a big chance it gets swallowed up by things like kingdom of miscellania approval notifications on login

Reply:
Done in https://github.com/pajlads/DinkPlugin/pull/429/commits/9b7a85e333d8966149383c446737ae8e65e0b694